### DISCUSSION TITLE
StochasticTools - saving solution for each sample

### DISCUSSION THREAD

**Discussion by kopanicakova** [idaholab/moose#17895] [Q&A Modules: General] [answered]
Hello, 

I am following parameter study example: 
https://mooseframework.inl.gov/modules/stochastic_tools/examples/parameter_study.html

In my sub app, I added following blocks: 
[VectorPostprocessors]
    [./save_var]
    type = NodalValueSampler
    variable = T
    sort_by = id
  [../]
[]

[Outputs]
  file_base = diffusion
  exodus = true
  csv = true
  execute_on = 'final'
[]

Basically, I would like to save solution of each sample to csv/exodus file. 
However, both exodus and csv files are overwritten for each sample. 
Is there any easy way to enable saving solution for each sample? 
Alternatively, is there a way to change name of file_base from master app? 

many thanks, 
Alena

**Comment by aeslaughter** [comment, marked as answer]
How many samples are you running? If you are not running a lot then you can set the `mode = normal` in the MultiApps block.

Another option is to transfer the data back from sub-application to the main.
https://github.com/idaholab/moose/blob/next/modules/stochastic_tools/test/tests/transfers/sampler_reporter/main.i

The [Transfers] and [Reporters] blocks are what you need. The "from_reporters" will need to be set to the value of the VPP vector that you want to save: from_reporter = save_var/{something}, the "something" will be the name that is showing up the CSV output now.

You will also need to enable JSON output, as such you will get your results in a JSON files instead of CSV files.
